vendor/k8s.io/kubernetes/pkg/api/rest/resttest/resttest.go

```diff
@@ -694,11 +694,16 @@ func (t *Tester) testUpdateRejectsMismatchedNamespace(obj runtime.Object, create
 		t.Errorf("unexpected error: %v", err)
 	}
 
-	objectMeta := t.getObjectMetaOrFail(obj)
+	storedFoo, err := getFn(ctx, foo)
```

Comment:
This is an upstream fix, see https://github.com/kubernetes/kubernetes/pull/40080. Basically, due to returning `false` from `AllowUncoditionalUpdate` all over our storage layer, we couldn't use `TestUpdate` method from the testing package. This is fixing that situation, so that it's modifying an existing object from a storage and not the one we submitted, which does not have proper `resourceVersion` value set, thus leading to failed updates.